cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerService.java

```diff
@@ -17,28 +17,40 @@
 package io.cdap.cdap.internal.app.worker.sidecar;
 
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.AbstractScheduledService;
 import com.google.inject.Inject;
 import io.cdap.cdap.common.conf.CConfiguration;
 import io.cdap.cdap.common.conf.Constants;
 import io.cdap.cdap.common.http.CommonNettyHttpServiceBuilder;
+import io.cdap.cdap.common.utils.DirUtils;
+import io.cdap.cdap.common.utils.FileUtils;
 import io.cdap.http.NettyHttpService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.io.IOException;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
 
 /**
  * Launches an HTTP server for receiving and unpacking and caching artifacts.
  */
-public class ArtifactLocalizerService extends AbstractIdleService {
+public class ArtifactLocalizerService extends AbstractScheduledService {
```

Comment:
Why should this be an http service? Would it serve any endpoints? 

One of the options to implement this cleaner could be 
```
executorService = Executors.newSingleThreadExecutor(Threads.createDaemonThreadFactory("artifact-cache-cleaner"));
```
and have  a runnable implementing cleaner.
https://github.com/cdapio/cdap/blob/develop/cdap-watchdog/src/main/java/io/cdap/cdap/logging/logbuffer/cleaner/LogBufferCleaner.java

You can start this scheduled service along with the service that creates these cached dirs.

Reply:
done